### ISSUE TITLE
how to limit the number of cores used by deepchem? [feature request]

### ISSUE THREAD

**Issue by UnixJunkie** [deepchem/deepchem#1539]
Hello,

It would be nice to have a non programmatic way to limit the max number
of cores used at run-time.
For example:
```
DEEPCHEM_MAX_THREADS=1 ./my_deepchem_script.py
```
Thanks,
F.

**Comment by peastman** [comment]
In the current version of TensorFlow, the way you limit the number of cores is with the ConfigProto you pass to the Session's constructor.  See https://stackoverflow.com/questions/38187808/how-can-i-run-tensorflow-on-one-single-core for details.  TensorGraph has a `configproto` argument you can use to specify that.

This is going to change in TensorFlow 2, though, since Sessions won't exist anymore.  I've been looking through the docs, but I haven't figured out what the new way of setting it will be.

**Comment by UnixJunkie** [comment]
Having an environment variable being taken into account would be nice.
Like OMP_NUM_THREADS for OpenMP.

**Comment by rbharath** [comment]
Is there still interest in limiting the number of cores for DeepChem?

I'm doing planning now and trying to figure out whether this is a feature we should support, so more details on your application would be helpful @UnixJunkie :)

**Comment by UnixJunkie** [comment]
My use case was to run several computations on a single computer.
Limiting the number of cores for each model being trained improves
the speedup.
In the extreme case, you want each model to use just one core, and you train
in parallel as many models as there are cores on the machine.

**Comment by UnixJunkie** [comment]
Answer to self; on Linux:
```bash
taskset -c 0 ./my_script.py
```
Will force execution to stay on CPU with identity 0.
